### ISSUE TITLE
Compile time DI exception

### ISSUE THREAD

**Issue by mgosk** [swagger-api/swagger-play#62]
I have problem when I'am trying attach Swagger to Play Framework app. Swagger lib scan uninitialized classes and cause problems. Any advices how to deal with it?

I extracted part of app as example:

https://github.com/mgosk/play-swagger-example

```
  CreationException: Unable to create injector, see the following errors:

    1) Error injecting constructor, java.lang.RuntimeException: There is no started application
      at auth.services.AuthService.<init>(AuthService.scala:24)
      while locating auth.services.AuthService
        for parameter 0 at auth.AnonymousAuthController.<init>(AuthController.scala:16)
      while locating auth.AnonymousAuthController
        for parameter 1 at router.Routes.<init>(Routes.scala:43)
      while locating router.Routes
      while locating play.api.inject.RoutesProvider
      while locating play.api.routing.Router
        for parameter 1 at play.modules.swagger.SwaggerPluginImpl.<init>(SwaggerPlugin.scala:33)
      while locating play.modules.swagger.SwaggerPluginImpl
      at play.modules.swagger.SwaggerModule.bindings(SwaggerModule.scala:11):
    Binding(interface play.modules.swagger.SwaggerPlugin to ConstructionTarget(class play.modules.swagger.SwaggerPluginImpl) eagerly) (via modules: com.google.inject.util.Modules$OverrideModule -> play.api.inject.guice.GuiceableModuleConversions$$anon$1)
    while locating play.modules.swagger.SwaggerPlugin
```

**Comment by Antanukas** [comment]
You could try defining vals lazy

**Comment by mrserverless** [comment]
I'm getting this error too. It's failing on the injection of `play.modeul.swagger.SwaggerPluginImpl` which is part of the library, not a developer class. 

I tried the workarounds suggested #43 by using the latest snapshot builds but didn't work either.

**Comment by elkhyary** [comment]
I'm having the same problem with the injection of `play.modules.swagger.SwaggerPluginImpl`.

**Comment by bantwald** [comment]
I am having the same issue with the following dependencies -
    "io.swagger" % "swagger-core" % "1.5.8",
    "io.swagger" %% "swagger-play2" % "1.5.1",
    "io.swagger" %% "swagger-scala-module" % "1.0.2-SNAPSHOT"

**Comment by frantuma** [comment]
Possibly related to #58, solved in #64  and version 1.5.2; please refer to https://github.com/swagger-api/swagger-samples/tree/master/scala/scala-play2.4 for sample usage

Please reopen if you continue to get issues.
